tests/functional/search-auth.js

```diff
@@ -198,6 +198,48 @@ registerSuite("Search (auth)", {
         .end();
     },
 
+    "Label have correct URL after search without results"() {
+      // load up a garbage search, so we can more easily detect when
+      // the search values we want are loaded.
+      return FunctionalHelpers.openPage(
+        this,
+        url(
+          "/issues",
+          "?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc"
+        ),
+        ".grid"
+      )
+        .findByCssSelector("#js-SearchForm-input")
+        .clearValue()
+        .click()
+        .type("noExpectedResult123")
+        .submit()
+        .getProperty("value")
+        .then(function(text) {
+          assert.equal(
+            text,
+            "noExpectedResult123",
+            "The q param populated the search input."
+          );
+        })
+        .end()
+        .findByCssSelector("#x-search-bar")
+        .submit()
+        .end()
+        .sleep(2000)
+        .findDisplayedByCssSelector(".label-browser-android")
+        .click()
+        .end()
+        .getCurrentUrl()
+        .then(function(url) {
+          assert.equal(
+            url,
+            "http://localhost:5000/issues?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc&q=label%3Abrowser-android",
```

Comment:
I *think* it might be possible for this to fail if someone passed in a new `--siteRoot`, other than `http://localhost:5000/`. I'll follow a file up bug, because it's hardcoded in other tests.

Maybe a safer option here would be `assert(url.includes("q=label%3Abrowser-android"))`, since that's the interesting bit of the test.

Reply:
👍

Reply:
I'll add this to this test, so we have less work later on.